src/WebJobs.Script.WebHost/Controllers/InstanceController.cs

```diff
@@ -35,11 +36,34 @@ public InstanceController(IEnvironment environment, IInstanceManager instanceMan
         [HttpPost]
         [Route("admin/instance/assign")]
         [Authorize(Policy = PolicyNames.AdminAuthLevel)]
-        public async Task<IActionResult> Assign([FromBody] EncryptedHostAssignmentContext encryptedAssignmentContext)
+        public async Task<IActionResult> Assign([FromBody] HostAssignmentRequest hostAssignmentContext)
         {
-            _logger.LogDebug($"Starting container assignment for host : {Request?.Host}. ContextLength is: {encryptedAssignmentContext.EncryptedContext?.Length}");
+            if (string.IsNullOrEmpty(hostAssignmentContext.EncryptedContext) &&
+                hostAssignmentContext.AssignmentContext is null)
+            {
+                return BadRequest("At least one of 'assignmentContext' or 'encryptedContext' must be provided.");
+            }
+            if (!string.IsNullOrEmpty(hostAssignmentContext.EncryptedContext) &&
+                hostAssignmentContext.AssignmentContext is not null)
+            {
+                return BadRequest("Only one of 'assignmentContext' or 'encryptedContext' may be set.");
+            }
+            if (!string.IsNullOrEmpty(hostAssignmentContext.EncryptedContext))
+            {
+                _logger.LogDebug("Starting container assignment. ContextLength is {ContextLength}", hostAssignmentContext.EncryptedContext.Length);
+            }
+            else
+            {
+                if (User != null &&
```

Comment:
I would not check for `User != null`. While it will never be null, because we require admin auth (which then enforces user is authenticated and will exist), we still want to be clear in this code we expect it.

I would just check the claim and let it throw if User isn't present (as it is an unexpected case and is a true `500`).

```suggestion
                if (!User.HasClaim(c => c.Type == SecurityConstants.AssignUnencryptedClaimType && c.Value == "true"))
```

Reply:
The user != null is needed for UTs. Where we are calling assign directly: https://github.com/Azure/azure-functions-host/blob/dev/test/WebJobs.Script.Tests.Integration/Management/InstanceControllerTests.cs#L90

Reply:
You can add a user to the test by assigning `instanceController.ControllerContext`

``` csharp
DefaultHttpContext context = new() { User = <create user> };
instanceController.ControllerContext = new() { HttpContext = context };
```

Reply:
Updated